### ISSUE TITLE
Live update of dimension values in N-panel during editing

### ISSUE THREAD

**Issue by EleotleCram** [EleotleCram/MeshTools#1]
Currently the dimension values in the N-panel update upon selection change, it would be useful to have these update as the mesh changes.

Technical details:

We can update the values as often as we like. The problem is, that our callback gets called very often (every time the 3D viewport updates).
Calculating the bounds on every callback, will overload the system. We need to come up with a good detection for when the values needs updating.

**Comment by 3DManuH** [comment]
Hi Marcel, 

Looking forward if there is a way to get live dimensions in edit view as i said.

- Somehow view3d.set_dimensions_mt doesn't work anymore when in edit mode!?
- Currently it just takes the pivot only. Can you add support to scale from cursor position as well?
- One more idea: Add option to not alter the scale when changing dimensions. So it always stays 1/1/1. Currently workflow is i change dimension and afterwards always have to ctrl-a apply scale. 

Cheers,
Manu

**Comment by EleotleCram** [comment]
Hi Manu,

Thanks for your feedback and suggestions.  Could you please create separate tickets for these items, so that they may be tracked individually? At the top-right of this page there should be a green button "New Issue". First two bullet points are 1 issue, the third is another.

Thanks in advance!

Marcel

**Comment by EleotleCram** [comment]
Hi Manu,

I have good news and bad news... 

The bad news is that I have decided to also abandon MeshTools addons. I have reverted my changes, so the N-Panel is removed again.

The good news is that I have created a new addon: https://github.com/EleotleCram/blender-cad-tools#cad-mesh-dimensions
This one has live update on selected mesh elements for meshes upto 10k verts. The new addon can be run alongside old Mesh Tools. (I would revert to the previous version, though)

Cheers,

Marcel
